docs: add adding react components doc (#18651)
gatsbyjs/gatsby: docs/docs/porting-from-create-react-app-to-gatsby.md

```jsx
if (typeof window !== `undefined`) {
  // code that references a browser global
  window.alert("Woohoo!")
}
```

> 2. (if you use class components) moving references to browser globals into a `componentDidMount`:

Reviewer: ```suggestion
2. For class components: moving references to browser globals into a `componentDidMount`
```

```jsx

import React, { Component } from "react"

class MyComponment() extends Component {
  render() {
    window.alert("This will break the build")

    return (
      <div>
        <p>Component</p>
      </div>
    )
  }
}
```

Would be changed into:

```jsx

import React, { Component } from "react"

class MyComponment() extends Component {
  componentDidMount() {
    // code that references the browser global
    window.alert("This won't break the build")
  }

  render() {
    return (
      <div>
        <p>Component</p>
      </div>
    )
  }
}
```

> 3. (if you use function components) moving references to browser globals into a `useEffect` hook:

Reviewer: ```suggestion
3. For function components: moving references to browser globals into a `useEffect` hook
```

export default Foo
```

> If these browser globals aren't protected correctly, you'll see a Webpack error like the one below when building your site:

Reviewer: ```suggestion
If these browser globals aren't protected correctly, you'll see a webpack error like the one below when building your site:
```
